nodes/explorer/src/api_backend/store.rs

```diff
@@ -44,3 +46,184 @@ where
         .collect();
     make_request_and_return_response!(futures::future::try_join_all(results))
 }
+
+type Depth = usize;
+
+#[derive(Deserialize)]
+pub(crate) struct BlocksByIdQueryParams {
+    from: BlockId,
+    to: Option<BlockId>,
+}
+
+pub(crate) async fn blocks<Tx, S>(
+    State(store): State<OverwatchHandle>,
+    Query(query): Query<BlocksByIdQueryParams>,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let BlocksByIdQueryParams { from, to } = query;
+    // get the from block
+    let from = match storage::block_req::<S, Tx>(&store, from).await {
```

Comment:
Seems that api_backend http part should only validate the query and params encoding, but the storage should check the existence of such ID. I think we have such separation of responsibilities in the nomos api.

Reply:
When making a query it probably returns an error, so yes, this may be redundant indeed.

Reply:
In our current storage implementation and trait definition, if a key is not found, then it will return a `None`, not an error. So I think we have to check if `from` exists here.